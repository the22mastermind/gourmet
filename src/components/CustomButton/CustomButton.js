import React from 'react';
import PropTypes from 'prop-types';
import { Dimensions, StyleSheet } from 'react-native';
import { Button, useTheme } from 'react-native-paper';

const {height} = Dimensions.get('screen');

const styles = StyleSheet.create({
  buttonContent: {
    height: height / 15,
  },
});

const CustomButton = ({testID, label, modeValue, onPress, loading}) => {
  const { colors } = useTheme();

  return (
    <Button
      testID={testID}
      accessibilityLabel={testID}
      mode={modeValue}
      onPress={onPress}
      loading={loading}
      contentStyle={styles.buttonContent}
      labelStyle={modeValue === 'text' ? { color: colors.link } : { color: colors.buttonText }}>
      {label}
    </Button>
  );
};

CustomButton.defaultProps = {
  loading: false,
};

CustomButton.propTypes = {
  testID: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  modeValue: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

export default CustomButton;
