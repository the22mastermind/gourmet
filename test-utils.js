import React from 'react';
import { render } from '@testing-library/react-native';
import PaperProvider from 'react-native-paper/lib/commonjs/core/Provider';

const AllTheProviders = ({ children }) => (
  <PaperProvider>{children}</PaperProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
