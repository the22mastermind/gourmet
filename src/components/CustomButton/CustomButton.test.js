import React from 'react';
import { render, fireEvent, act } from 'test-utils';
import CustomButton from './CustomButton';

describe('<CustomButton /> component', () => {
  it('renders <CustomButton /> correctly', () => {
    const onPress = jest.fn();
    const screen = render(
      <CustomButton
        testID="my custom button"
        label="my custom button"
        modeValue="contained"
        onPress={onPress}
        loading={false}
      />);
  
    expect(screen.getByText('my custom button')).toBeTruthy();
  });
  
  it('calls onPress event correctly', async () => {
    const onPress = jest.fn();
    const screen = render(
      <CustomButton
        testID="my custom button"
        label="my custom button"
        modeValue="contained"
        onPress={onPress}
        loading={false}
      />);
    
    await act(async () => {
      fireEvent.press(screen.getByText('my custom button'));
    });
  
    expect(screen.getByText('my custom button')).toBeTruthy();
    expect(onPress).toHaveBeenCalledTimes(1);
  });
  
  it('matches <CustomButton /> component snapshot', async () => {
    const onPress = jest.fn();
    const screen = render(
      <CustomButton
        testID="my custom button"
        label="my custom button"
        modeValue="contained"
        onPress={onPress}
        loading={false}
      />);
  
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
