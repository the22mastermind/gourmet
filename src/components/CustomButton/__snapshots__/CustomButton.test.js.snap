// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> component matches <CustomButton /> component snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#a4c49a",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityLabel="my custom button"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="my custom button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 88.93333333333334,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#ffffff",
                },
              ],
            ]
          }
        >
          my custom button
        </Text>
      </View>
    </View>
  </View>
</View>
`;
